Realm/RLMSyncSubscription.mm

```diff
@@ -0,0 +1,528 @@
+
+
+- (NSUInteger)count {
+    return _subscriptionSet->size();
+}
+
+- (nullable NSError *)error {
+    NSString *errorMessage = RLMStringDataToNSString(_subscriptionSet->error_str());
+    if (errorMessage.length == 0) {
+        return nil;
+    }
+    return [[NSError alloc] initWithDomain:RLMFlexibleSyncErrorDomain code:0 userInfo:@{NSLocalizedDescriptionKey: errorMessage}];
+}
+
+- (RLMSyncSubscriptionState)state {
+    switch (_subscriptionSet->state()) {
+        case realm::sync::SubscriptionSet::State::Uncommitted:
+        case realm::sync::SubscriptionSet::State::Pending:
+        case realm::sync::SubscriptionSet::State::Bootstrapping:
+            return RLMSyncSubscriptionStatePending;
+        case realm::sync::SubscriptionSet::State::Complete:
+            return RLMSyncSubscriptionStateComplete;
+        case realm::sync::SubscriptionSet::State::Error:
+            return RLMSyncSubscriptionStateError;
+        case realm::sync::SubscriptionSet::State::Superceded:
+            return RLMSyncSubscriptionStateSuperseded;
+    }
+}
+
+#pragma mark - Batch Update subscriptions
+
+- (void)write:(__attribute__((noescape)) void(^)(void))block {
+    return [self write:block onComplete:^(NSError*){}];
+}
+
+- (void)write:(__attribute__((noescape)) void(^)(void))block
+   onComplete:(void(^)(NSError *))completionBlock {
+    if (self->_isInWriteTransaction) {
+        @throw RLMException(@"Cannot initiate a write transaction on subscription set that is already been updated.");
+    }
+    _mutableSubscriptionSet = std::make_unique<realm::sync::MutableSubscriptionSet>(_subscriptionSet->make_mutable_copy());
+    self->_isInWriteTransaction = true;
+    block();
+    try {
+        _subscriptionSet = std::make_unique<realm::sync::SubscriptionSet>(std::move(*_mutableSubscriptionSet).commit());
+        _mutableSubscriptionSet = nullptr;
+        self->_isInWriteTransaction = false;
+    }
+    catch (const std::exception& error) {
+        NSError *err = [[NSError alloc] initWithDomain:RLMFlexibleSyncErrorDomain code:-1 userInfo:@{@"reason":@(error.what())}];
+        return completionBlock(err);
+    }
+    _subscriptionSet->get_state_change_notification(realm::sync::SubscriptionSet::State::Complete)
+        .get_async([completionBlock](realm::StatusWith<realm::sync::SubscriptionSet::State> state) mutable noexcept {
+            if (state.is_ok()) {
+                completionBlock(nil);
+            } else {
+                NSError* error = [[NSError alloc] initWithDomain:RLMFlexibleSyncErrorDomain code:state.get_status().code() userInfo:@{@"reason": @(state.get_status().reason().c_str())}];
+                completionBlock(error);
+            }
+        });
+}
+
+typedef void(^RLMSyncSubscriptionCallback)(NSError * _Nullable error);
+
+- (BOOL)writeAsync:(__attribute__((noescape)) void(^)(void))block
+          callback:(RLMSyncSubscriptionCallback)callback {
+    [NSException raise:@"NotImplemented" format:@"Needs Implementation"];
```

Comment:
Do we plan to implement this for the initial release?

Reply:
This is not included on the scope for this first beta release

Reply:
We should just omit this for now and add it back on the next version on flex sync